qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/commons/QuteTemplateCodeActionParams.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package com.redhat.qute.commons;
+
+/**
+ * Parameters for a request from qute-ls to the JDT component to resolve a
+ * CodeAction that requires knowledge of the Java project.
+ *
+ * @author datho7561
+ */
+public class QuteTemplateCodeActionParams {
```

Comment:
I would not try to do a generic codeaction for template in jdt side because your code action parameters could grow with new parameters like deleteJavaFileUri to delete a java file.  It will be hard to unterstand your code action parameters. 

I suggest that you keep your parameyer like this but rename your class like GenerateJavaMemberParameters ans replace in jdt side codeaction with your new name. More on jdt you return à workspaceEdit and not à codeaction so it is better to use a name like GenerateJavaMember instead of codeaction. Fell free to use ana noter name but more specific than codeaction.